packages/react/react/src/hooks/usePreservedReference.spec.tsx

```diff
@@ -0,0 +1,67 @@
+import { renderHook } from '@testing-library/react';
+import { useEffect, useState } from 'react';
+import { useCallbackOnce } from './useCallbackOnce';
+import { usePreservedReference } from './usePreservedReference';
+
+type TossObject = {
+  toss: string;
+};
+
+describe('usePreservedReference', () => {
+  const callbackReturnTrue = jest.fn(() => true);
+  const callbackReturnFalse = jest.fn(() => false);
+  const tossObject: TossObject = { toss: '토스' };
+
+  const useTestCustomHook = (object: TossObject, areValuesEqual?: () => boolean) => {
```

Comment:
type of `areValuesEqual ` in this file is `() => boolean`
but in implementation of usePreservedReference hook areValuesEqual type is `(a: T, b: T) => boolean`

Is there no problem if the two types are different?😄

Reply:
I thought it wouldn't be a problem!

The reason is that when `customFunction` is assigned to `areValuesEqual`, the expected behavior of `customFunction` is to return `true` if the **values are equal**, and `false` if the **values are different**.

Thank you for the review!😀

Reply:
Thank you : )